### ISSUE TITLE
Consider allowing numbers that fail the StreamReadConstraints limits to be truncated

### ISSUE THREAD

**Issue by pjfanning** [FasterXML/jackson-core#915]
* probably an additional param on StreamReadConstraints that controls whether a number that is too long leads to an exception or if we truncate the least significant digits
* need to bear in mind that numbers may be in 1.23 format (truncate left most chars) or on in 1.23e987 format where we need to retain the 'e987' but truncate the chars to left of the 'e'.
* Jackson-core is used by many libs and frameworks and they typically don't allow much configurability of the ObjectMappers they create. The v2.15.0 change to fail when numbers are longer than 1000 chars (or bytes, depending on the context) could lead to real problems for those libs/frameworks.

**Comment by cowtowncoder** [comment]
I am not sure this is practical for floating-point numbers for stated reasons?

And I guess that although integers are easier, you would probably never want to truncate those -- whereas with FP you might want to drop trailing digits (even if tricky), you would not want to truncate anything from integers as magnitude always changes.

**Comment by pjfanning** [comment]
* reasonable points - truncation algorithm won't be easy to get right and may need to fail for some cases
* maybe best to simply doc the facts about the new exceptions and to let the lib maintainers worry about how to react

**Comment by cowtowncoder** [comment]
Yes. Would make sense to try to engage maintainers of frameworks like Quarkus, Spring Boot, RESTeasy, DropWizard et al so perhaps they could help beta-test changes.

It is tricky business to select good limits.

Plus I guess there is still one thing that we have discussed a few times but that I wonder about... maximum number length limits for floating-point numbers (where 768 or 1000 seem upper bounds for DoS/performance) vs those for integrals (even `BigInteger` should be able to handle much longer than 1000 digits without too much trouble).
Is there any benefit from making integer-limit looser?

As to documentation, wonder what would be best ways to do that? I added some verbiage now on:

https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15

but there must be other places to consider.
One thing to maybe consider would be adding something in exception messages `StreamReadConstraints` validation methods use -- perhaps reference to `StreamReadConstraints` setting itself?
